### PR TITLE
TST: Also run slow test in non-dev env, fix Catalog Search bug

### PR DESCRIPTION
**Pull request by pllim**
Probably an oversight to only run slow tests on devdeps. I bet it would also fail on stable astropy but let's see. Yes, it did. I bet the code never worked in the first place. It was trying to append rows with incompatible data types.

[🐱](https://jira.stsci.edu/browse/JDAT-4677)

### CODE REVIEW

File: jdaviz/configs/imviz/plugins/catalogs/catalogs.py
Comment (kecnry) [reply jdaviz/configs/imviz/plugins/catalogs/catalogs.py:1]: > It was trying to append rows with incompatible data types.

@pllim - does this mean that the test case had some entries where the Object ID was populated (as an integer) and so then additional rows had to be castable to that same type?  Is Object ID always going to be integer by default, couldn't it be a string for some catalogs?

Comment (pllim) [reply jdaviz/configs/imviz/plugins/catalogs/catalogs.py:1]: I don't have definite answers for all the questions. I only know from the SDSS catalog queried in test. In that case, ID is always integer, so appending empty string crashes it. Also there is no reason to append dummy value in the first place because you should really write out the ID numbers as well to be read back in; otherwise the catalog is useless because you cannot tie the dots to anything meaningful anymore. Does that make sense or you have more concerns?

Comment (kecnry) [reply jdaviz/configs/imviz/plugins/catalogs/catalogs.py:1]: If the currently implemented catalogs always have Object ID as an integer, then I think this makes sense for now.  Do you know if that's the case @kcarver1?

**Review by haticekaratay — APPROVED** [review]
Approving with the assumption that ObjectID is an integer. This does the fix! Thanks @pllim

File: jdaviz/configs/imviz/plugins/catalogs/catalogs.py
Comment (kecnry) [reply jdaviz/configs/imviz/plugins/catalogs/catalogs.py:1]: This will also restrict us from _appending_ catalogs from file in which the Object ID is not an integer, is the current behavior to clear the catalog when loading from a file (after doing an online query)?  If not, maybe we should change to that for now unless we're requested to support appending, in which case we'll need to figure out a way to handle the type mismatches - perhaps by always casting to a string instead.

Comment (pllim) [reply jdaviz/configs/imviz/plugins/catalogs/catalogs.py:1]: Well, if you want to keep it as string, then more code changes are needed to always cast ID to string, so in SDSS's case, the query returns ID as int, but then when user writes it out, it will always be string. Do you want that instead?

Comment (pllim) [reply jdaviz/configs/imviz/plugins/catalogs/catalogs.py:1]: Looks like the user doc might also need updating because the code looks for "label" but it is not mentioned at all in the user doc. But I will wait for your reply before I make more changes. Thanks!

Comment (pllim) [reply jdaviz/configs/imviz/plugins/catalogs/catalogs.py:1]: Handled in the latest commit.

**Comment by pllim** [comment]
@haticekaratay , ID is now always string instead of integer. Do you want to re-review? Thanks!

File: docs/imviz/plugins.rst
Comment (kecnry) [inline docs/imviz/plugins.rst:386]: `label` is optional now, right?  And won't the logic here cast a column of integers to a string, or no?

Comment (pllim) [reply docs/imviz/plugins.rst:386]: In practice, I don't think a catalog is useful without identifiers. We should encourage users to pass them in.

Comment (pllim) [reply docs/imviz/plugins.rst:386]: The existing "From File" logic is pretty bare-bones. All I see is:

```python
                row_info = {'Right Ascension (degrees)': row['sky_centroid'].ra.deg,
                            'Declination (degrees)': row['sky_centroid'].dec.deg,
                            'Object ID': row.get('label', 'N/A')}
```

So, no, it does not turn input int column into string.

Comment (pllim) [reply docs/imviz/plugins.rst]: But I added a commit to fix that as well. Please re-re-review. Thanks!

**Review by kecnry — APPROVED** [review]
Thanks!

**Pull request merged**
